### PR TITLE
Ports tgui-next 3.0, removes tgui 1.0 + Smart Asset Cache

### LINKED ISSUE #7651: Can no longer manually set SMES'
Have to use the buttons like some animals.

### LINKED ISSUE #8885: Opening Cyborg Remote Monitoring Bricks Tablet Computers.
## Round ID:
30017

## Testmerges:
#8291: 'Ports tgui-next 3.0, removes tgui 1.0 + Smart Asset Cache' by TheGamerdk at commit ed55b235d6
#8598: 'Rebalances Security Grenades' by MacHac at commit ea19643320
#8864: 'Memory Profiler for Server use' by JamieD1 at commit 4002456f1f

## Reproduction:
I had downloaded Cyborg Remote Monitoring on my Tablet Computer (The one HoP recieves roundstart). I do not recall what other applications were installed. Upon opening said application, the screen read "Interface ntos_cyborg_monitor was not found." There was no way that the application could be closed. The window itself could be closed, but this made all other applications inaccessible, which sucked as I had downloaded a number of other programs. The mentor Xoxeyos tried to remedy this, but there was no way that the application could be "refreshed."

I was told that we did infact have a cyborg, named G.U.B, so the lack of borgs may not have caused this.

### PR DESCRIPTION
**Pull request by Bibby0110**
### IMPORTANT. THIS RELICENSES TGUI TO MIT TO MIRROR TG

This adds a TGUI for:
The teleporter
The timer
The mining vendor
The infrared thing
The proximity sensor
The radioactive microlaser
Emergency Shuttle UI
AI integrity restorer
Intellicard
Robotics console
Exosuit console
Preferences menu
Particle Accelerator
Mulebots
Slime body swapper
Signaller
Bluespace Launchpad + Remote
Eightball
NTNet relay
Labor Claim Console
Solar tweaks
Contractor Hub port + UI
Various NTOS programs
DNA console for genetics
ByondUI including a new camera console UI
Pirate Cargo Hold
AI Robot Control
Malf ability picker
Electropack
TTV
Fixes Canvas and gives a TGUI
Canister UI
Biogenerator
Seed extractor
Nanite Machines
The vending machines
Crew Monitoring Console

**TODO:**
- [x] Clockwork slab fix
- [x] Autolathe Fix
- [x] Crew Console
- [x] Remove tgui-old
- [x] Regain my sanity
- [x] Psi Web?

Also:
Solar panel refactor (Visuals only)
Contractor Hub (See https://github.com/tgstation/tgstation/pull/45117)
Fixes a small bug in the uplink
Makes it possible to switch to nightshift while an APC is locked
Adds a number input for the SMES input/output
Fixes cargo tooltips, it now shows what's in the crate as a tooltip
The Captain no longer has to pay for his CapDrobe
Fixes the Codex Gigas

Ports Knobs & Sliders from TG
Removes old TGUI.
Ports TGUI-3.0. (Removal of routes.js)

Autolathe fixes. 
See https://github.com/yogstation13/Yogstation/pull/8770
Closes #8770 

ALSO:
Ports the Smart Asset Cache from TG
https://github.com/tgstation/tgstation/pull/49982
TL;DR: Assets are no longer resent each time you reconnect after a server reboot.

ALSO:
Removes the rastabeanie, it had no sprite.

Fixes #7651 
Fixes #8885

#### Changelog

:cl:  TheGamerdk for porting, original authors for coding
rscadd: Various new TGUI's from TGstation
/:cl:

### CODE REVIEW

**Comment by Bibby0110** [comment]
oh god I forgot to compile it

**Comment by Firewolf34** [comment]
I love you for this.

**Comment by Bibby0110** [comment]
> I love you for this.

I'm not into that

**Review by JamieD1 — CHANGES_REQUESTED** [review]
Conflicts

**Review by JamieD1 — APPROVED** [review]

File: code/modules/vending/_vending.dm [hunk 1]
```diff
@@ -350,196 +350,160 @@ GLOBAL_LIST_EMPTY(vending_products)
...
+	. = list()
 	var/mob/living/carbon/human/H
 	var/obj/item/card/id/C
-	//yogs start -- ignores_capitalism stuff
+	
 	var/mob/living/L
 	if(isliving(user))
 		L = user
-	//yogs end
+	.["ignores_capitalism"] = L.ignores_capitalism
```
Comment (SomeguyManperson) [inline code/modules/vending/_vending.dm]: this runtimes for ghosts

**Comment by SomeguyManperson** [comment]
I don't know what this means but it's a runtime so
image

File: code/modules/vending/_vending.dm (hunk 1 above)
Comment (Bibby0110) [reply code/modules/vending/_vending.dm]: Ghosts are mob/living?

**Comment by Bibby0110** [comment]
It means our codebase is fucked. They’re items with missing Icons. I’ll link some of the fix PRs

File: code/modules/vending/_vending.dm (hunk 1 above)
Comment (Bibby0110) [reply code/modules/vending/_vending.dm]: Oh, nevermind

**Comment by Bibby0110** [comment]
> I don't know what this means but it's a runtime so
> image

see above

EDIT: I forgot to link the PRs

**Comment by JamieD1** [comment]
You could of just reversed it

**Comment by Bibby0110** [comment]
> You could of just reversed it

wat

**Comment by patpol4** [comment]
What did the rastabeanie ever do to you

**Comment by Bibby0110** [comment]
> What did the rastabeanie ever do to you

Live without a sprite

**Comment by Bibby0110** [comment]
Updated the description because SOME people thought we had undocumented changes

**Review by SomeguyManperson — APPROVED** [review]
